### DISCUSSION TITLE
Are properties always fetched?

### DISCUSSION THREAD

**Discussion by nk-coding** [ExpediaGroup/graphql-kotlin#1321] [Q&A] [answered]
Hi,  
the [documentation for fetching data](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/execution/fetching-data/) says, that fields backed by a property are always fetched, regardless whether or not asked for:  
> If we expose the reviews field as a property it will always be populated regardless whether or not your client actually asks for it.

However, when I added a breakpoint to a property getter, the program only ran in the breakpoint, if the query asked for that field.  
Therefore I want to ask if  
1. I can rely on that behavior (will the property getter only be called if it is queried for)
2. This is a bug/outdated information in the documentation (if so, it might be related to #1112, as this PR changed how properties are fetched, however I am really not sure there)

**Comment by dariuszkuc** [comment, marked as answer]
Hello 👋 
Yes data fetchers for corresponding fields (regardless whether they are backed by a function or a property) will only be called if given query explicitly asks for it.

Documentation simply states that field has to be somehow initialized, i.e. given schema

```graphql
type Foo {
  # this field is quickly resolved in milliseconds
  bar: String!
  # this field is backed by slow API and it takes 10 second to compute
  baz: Int! 
}
```

If you simply expose that type as a

```kotlin
data class Foo(val bar: String, val baz: Int)
```

then it is not possible to create an instance of class `Foo` without calculating the value of `bar` (fast) and `baz` (slow) fields. Regardless whether user asks for slow `baz` or not, we HAVE to calculate it before returning `Foo` from a function. On the other hand, if you expose `baz` as a function then that function will only be invoked if user explicitly asks for it.

```kotlin
class Foo(val bar: String) {
  fun baz(): Int {
     // some 
     // this code only executes if query asks for baz
  }
}
```

**Reply by nk-coding** [reply]
Thanks for the answer
